File: insecure-atm-sample.js
// Insecure ATM Sample Code for SonarCloud Testing
// This file intentionally contains vulnerabilities for academic scanning purposes.

// Hardcoded credentials (SonarCloud will flag this)
const ADMIN_PIN = "4567";

// Simulated ATM balance
let balance = 1000;

// Insecure logging of sensitive data
function logTransaction(userId, amount, pin) {
    console.log("User:", userId, "Amount:", amount, "PIN:", pin); // Sensitive info leak
}

// Dangerous use of eval()
function calculateFee(expression) {
    return eval(expression); // SonarCloud will flag eval()
}

// No input validation
function withdraw(userId, amount, pin) {
    if (pin !== ADMIN_PIN) {
        console.log("Invalid PIN");
        return;
    }

    // Insecure direct file write (SonarCloud will flag)
    const fs = require("fs");
    fs.writeFileSync("transaction_log.txt", `User ${userId} withdrew ${amount}`);

    logTransaction(userId, amount, pin);

    balance -= amount;
    console.log("Withdrawal successful. New balance:", balance);
}

// Insecure function call
withdraw("user001", 300, "4567");

// Dangerous fee calculation
console.log("Fee:", calculateFee("10 + 5 * 2"));

File: src/insecure-atm-sample.py
// Insecure ATM Sample Code for SonarCloud Testing
// This file intentionally contains vulnerabilities for academic scanning purposes.

import os

# Hardcoded credentials (SonarCloud will flag this)
ADMIN_PIN = "4567"

# Simulated ATM balance
balance = 1500

# Insecure logging of sensitive data
def log_transaction(user_id, amount, pin):
    print(f"[LOG] User: {user_id}, Amount: {amount}, PIN: {pin}")  # Sensitive info leak

# No input validation
def withdraw(user_id, amount, pin):
    global balance

    if pin != ADMIN_PIN:
        print("Invalid PIN")
        return

    # Insecure file write (SonarCloud will flag)
    with open("transaction_log.txt", "w") as f:
        f.write(f"User {user_id} withdrew {amount}\n")

    log_transaction(user_id, amount, pin)

    # No check for negative balance
    balance -= amount
    print("Withdrawal successful. New balance:", balance)

# Dangerous use of eval()
def calculate_fee(expression):
    return eval(expression)  # SonarCloud will flag eval()

# Insecure function call
withdraw("user123", 500, "4567")

# Dangerous fee calculation
print("Fee:", calculate_fee("10 + 5 * 3"))
